File: CS/ChartsDemo.Wpf/ViewModels/WeatherData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace ChartsDemo {
    [XmlRoot("WeatherData")]
    public class WeatherData : List<WeatherInfo> {
        static List<WeatherInfo> data = null;

        public static List<WeatherInfo> Data {
            get {
                if(data == null) {
                    var s = new XmlSerializer(typeof(WeatherData));
                    data = (List<WeatherInfo>)s.Deserialize(DataLoader.LoadFromResources("/Data/WeatherData.xml"));
                }
                return data;
            }
        }
    }
    public class WeatherInfo {
        public DateTime Timestamp { get; set; }
        public double Temperature { get; set; }
        public double Pressure { get; set; }
        public double RelativeHumidity { get; set; }
    }
}

File: VB/ChartsDemo.Wpf/ViewModels/WeatherData.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Xml.Serialization

Namespace ChartsDemo
    <XmlRoot("WeatherData")>
    Public Class WeatherData
        Inherits List(Of WeatherInfo)


        Private Shared data_Renamed As List(Of WeatherInfo) = Nothing

        Public Shared ReadOnly Property Data() As List(Of WeatherInfo)
            Get
                If data_Renamed Is Nothing Then
                    Dim s = New XmlSerializer(GetType(WeatherData))
                    data_Renamed = DirectCast(s.Deserialize(DataLoader.LoadFromResources("/Data/WeatherData.xml")), List(Of WeatherInfo))
                End If
                Return data_Renamed
            End Get
        End Property
    End Class
    Public Class WeatherInfo
        Public Property Timestamp() As Date
        Public Property Temperature() As Double
        Public Property Pressure() As Double
        Public Property RelativeHumidity() As Double
    End Class
End Namespace
